### DISCUSSION TITLE
Social Media Automation

### DISCUSSION THREAD

**Discussion by benfrankie** [seleniumbase/SeleniumBase#2735] [Q&A] [answered]
Hi Michael, just curious how Seleniumbase compares to commercial anti-detect browsers like Dolphin in terms of anti-detection effectiveness?

I've been using Seleniumbase for social media web automation, but my accounts consistently get banned within 24 hours, even with a reconnect time set between 15-30 seconds. Notably, Seleniumbase appears more effective at bypassing Cloudflare screens compared to Adspower, Gologin, and others. 

However, the bans tend to occur more frequently when I run more than 10 browsers simultaneously, despite implementing delays between actions. Additionally, some accounts get banned without any activity, even when using residential proxies. This leads me to suspect the issues may be related to fingerprinting. Any suggestions? 

Below is the relevant part of my code: 
            ua = UserAgent()
            user_agents = [ua.random for _ in range(20)]

            driver = Driver(
                uc=True,
                agent=random.choice(user_agents),      
                undetectable=True,        
                incognito=True,
                proxy=proxy_ip_port
                )

    driver.uc_open_with_reconnect(login_url, reconnect_time=30)
...

Thank you.

**Comment by mdmintz** [comment, marked as answer]
If the random user_agent you give it doesn't match with the rest of the browser, then it could be flagged as a bot. SeleniumBase's default user_agent gives it the optimal setting for making the browser appear as a normal human-controlled web browser.

Also, as mentioned in the video (https://www.youtube.com/watch?v=2pTpBtaE7SQ&t=1349s), unusual web traffic coming from an IP Address can get flagged, and then blocked. This is why setting the `proxy` option may be needed if running bots in parallel.

Not all proxy addresses are created equal. If you're reusing a proxy that was previously used by someone else, then that proxy address may have already been flagged... and blocked by certain websites.
